### PR TITLE
Fixed timezone bug in view and create

### PR DESCRIPTION
**Pull request by Gautime**
Changed the timezone settings in create command and also fixed the **UTC undefined** error in view event command.

### CODE REVIEW

File: helpers/GSGetter.ts [hunk 1]
```diff
@@ -257,18 +287,46 @@ export class GCGetter {
...
+                const inviteend_ms = invite_end_datetime.split(".");
+
+                const users_info = await read.getUserReader().getById(users_id);
+
+                let final_start;
+                let final_end;
+                let utcoffset = users_info.utcOffset;
+                let decimal_part = Math.abs(utcoffset - Math.floor(utcoffset));
+                decimal_part = decimal_part * 60;
+                if (utcoffset > 0) {
```
Comment (d-gubert) [inline helpers/GSGetter.ts]: This looks very similar to the code above, can you extract this logic into a function?

Comment (Gautime) [reply helpers/GSGetter.ts]: https://github.com/RocketChat/Apps.Google.Calendar/blob/Fix2/helpers/result.ts#L68
Made the required changes.

File: helpers/GSGetter.ts [hunk 2]
```diff
@@ -106,15 +106,15 @@ export class GCGetter {
                 const minimum_date = dat.toISOString();
                 const url = `https://www.googleapis.com/calendar/v3/calendars/primary/events?key=${api_key}&showDeleted=false&timeMin=${minimum_date}`;
                 let api_response = await http.get(url, { headers: { Authorization: `Bearer ${view_token}` } });
-
+                console.log('This is the view api response:', api_response);
```
[the remark this answers is not in the document]
Comment (Gautime) [reply helpers/GSGetter.ts]: Removed console logs from all the files.

File: helpers/result.ts [hunk 3]
```diff
@@ -76,3 +64,50 @@ export async function refresh_access_token(token: string, read: IRead, http: IHt
         console.log('Encountered error during refreshing access token:', refresh_response.data.error.message);
     }
 }
+
+export async function make_time_string( date: string, time: string, utc: number) : Promise<string>{
```
[the remark this answers is not in the document]
Comment (Gautime) [reply helpers/result.ts]: Updated the function.

File: helpers/result.ts [hunk 4]
```diff
@@ -76,3 +64,50 @@ export async function refresh_access_token(token: string, read: IRead, http: IHt
...
+
+    const date_string = date + 'T' + time + 'Z';
+    const new_date = new Date(date_string);
+    const datetime = new_date.toISOString();
+    const datetime_ms = datetime.split(".");
+
+    let final_time;
+    let decimal = Math.abs(utc - Math.floor(utc));
+    decimal = decimal * 60;
+    if (utc > 0) {
```
[the remark this answers is not in the document]
Comment (Gautime) [reply helpers/result.ts]: Formatted the function in a proper manner.

File: helpers/result.ts [hunk 5]
```diff
@@ -76,3 +64,50 @@ export async function refresh_access_token(token: string, read: IRead, http: IHt
         console.log('Encountered error during refreshing access token:', refresh_response.data.error.message);
```
[the remark this answers is not in the document]
Comment (Gautime) [reply helpers/result.ts]: Changed the error statement from console log to logger statements.

File: helpers/result.ts [hunk 6]
```diff
@@ -2,40 +2,29 @@ import { IModify, IRead, IHttp, HttpStatusCode, IPersistence } from '@rocket.cha
 import { SlashCommandContext, ISlashCommand } from '@rocket.chat/apps-engine/definition/slashcommands';
 import { AppPersistence } from '../helpers/persistence';
 
-export async function displayevents(result: any, modify: IModify, context: SlashCommandContext): Promise<void> {
+export async function displayevents(result: any, modify: IModify, context: SlashCommandContext, time: any): Promise<void> {
```
[the remark this answers is not in the document]
Comment (Gautime) [reply helpers/result.ts]: Updated the function name in definition and function calls.

**Comment by Gautime** [comment]
@d-gubert  @rodrigok Please review and merge this PR to fix the bugs in master branch.

**Pull request merged**